content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+
+SRGs can contain unbounded arrays for resource declarations: SRVs, UAVs, CBVs and Samplers. There are some rules that need to be considered, depending on command line arguments when compiling with **AZSLc**. More on unbounded arrays [here](unbounded-arrays).
+
+SRGs can contain functions.
+
+SRGs can contain structs (declaration & instances). The Shader Build Pipeline and the runtime handle the platform-specific packing rules and alignment.
+
+SRGs can contain instances of classes (but you can not declare classes inside an SRG).  
+  
+**Example**  
+The following example demonstrates declaration of a Shader Resource Group. We have not covered details of what is a `ShaderResourceGroupSemantic`, but suffice to say, it defines the space (aka, descriptor set) index of the constants and resources declared inside the SRG.  
+  
+    ShaderResourceGroupSemantic BindingPerExample
+    {
+        FrequencyId = 0;
+    };
+       
+    ShaderResourceGroup ExampleSRG : BindingPerExample
+    {
+        float4          m_uniformColor;
+        Texture2D       m_texture;
+        Buffer<float>   m_buffer;
+        Sampler m_dynamicSampler;
+        Sampler m_staticSampler
+        {
+            AddressU = Wrap;
+            AddressV = Wrap;
+            AddressW = Wrap;
+            MagFilter = Linear;
+        };
+        
+        // You can define functions inside a ShaderResourceGroup
+        float GetRedColor()
+        {
+            return m_uniformColor.r;
+        }
+    };
+  
+### Scope
+Every SRG defines a scope and the SRG data must be qualified when accessed from out-of-SRG scope code.  
+  
+    float4 color = ExampleSRG::m_uniformColor;
+    color *= ExampleSRG::m_texture.Sample(ExampleSRG::m_staticSampler, float2(0.5, 0.5));
+  
+
+### Sampler variables
+* Dynamic Samplers are declared without a body definition. They have to be set in the runtime. Dynamic Samplers can also be declared as arrays.
+* Static Samplers are declared with a body definition.
+    * Although you can declare Samplers directly as `SamplerState` or `SamplerComparisonState`, AZSL provides a convenient single `Sampler` keyword for both.
+    * Sampler variables will be compiled into regular HLSL `SamplerState` variables, unless a `ComparisonFunc` is defined inside the `Sampler` block, which will be compiled as `SamplerComparisonState`.  
+
+<!-- -->
+
+    //These samplers are declared inside an SRG
+    ...
+    Sampler m_dynamicSampler;
+     
+    // Compiled as HLSL SamplerState
+    Sampler m_staticSamplerState
+    {
+        AddressU = Wrap;
+        AddressV = Wrap;
+        AddressW = Wrap;
+        MagFilter = Linear;
+    };
+     
+    //Compiled as HLSL SamplerComparisonState because of the ComparisonFunc declaration.
+    Sampler m_staticSamplerComparisonState
+    {
+        AddressU = Wrap;
+        AddressV = Wrap;
+        AddressW = Wrap;
+        MagFilter = Linear;
+        ComparisonFunc = Less;
+    };
+    ...
```

Comment:
```suggestion
The following examples demonstrate declarations for a dynamic sampler, a static sampler, and a static sampler comparison. They are declared inside an SRG.

/```c
  // Dynamic sampler declaration.
  Sampler m_dynamicSampler;
   
  // Static sampler declaration. Compiles as HLSL SamplerState.
  Sampler m_staticSamplerState
  {
      AddressU = Wrap;
      AddressV = Wrap;
      AddressW = Wrap;
      MagFilter = Linear;
  };

  // Static sampler declaration. Compiles as HLSL SamplerComparisonState because of the ComparisonFunc definition.
  Sampler m_staticSamplerComparisonState
  {
      AddressU = Wrap;
      AddressV = Wrap;
      AddressW = Wrap;
      MagFilter = Linear;
      ComparisonFunc = Less;
  };
/```
```

Reply:
Fixed.